src/extension/extensions/CodeCompletionExtension.ts

```diff
@@ -113,9 +115,9 @@ export class CodeCompletionExtension extends BaseExtension implements Completion
             token.isCancellationRequested ||
             !index.indexReady ||
             (lineText.substring(0, position.character).match(/["'`]/g) || []).length % 2 === 1 ||
-            lineText.match(/^\s*(\/\/|\/\*\*|\*\/|\*)/g) ||
+            REGEX_COMMENT.test(lineText) ||
             lineText.startsWith('import ') ||
-            lineText.substring(0, position.character).match(new RegExp(`(\w*[.])+${searchWord}`, 'g'))) {
+            new RegExp(`(?:\w*\.)+${searchWord}`).test(lineText.substring(0, position.character))) {
```

Comment:
- Escaping is idiomatically done with backslashes (you usually do, btw)
- Switching to non-capturing group for better perf (as we won't be needing the group capture)
- Stripping useless costly `g` flag
- `regex.test(x)` perfs better than `x.match(regex)` when all we need is a yes/no